packages/data-api/src/TupaiaDataApi.ts

```diff
@@ -6,18 +6,24 @@
 import groupBy from 'lodash.groupby';
 
 import moment from 'moment';
+import { TupaiaDatabase } from '@tupaia/database';
 import { getSortByKey, DEFAULT_BINARY_OPTIONS } from '@tupaia/utils';
 import { SqlQuery } from './SqlQuery';
-import { AnalyticsFetchQuery } from './AnalyticsFetchQuery';
-import { EventsFetchQuery } from './EventsFetchQuery';
-import { sanitizeMetadataValue, sanitizeAnalyticsTableValue } from './utils';
+import { AnalyticsFetchOptions, AnalyticsFetchQuery } from './AnalyticsFetchQuery';
+import { EventsFetchQuery, EventAnswer, EventsFetchOptions } from './EventsFetchQuery';
+import {
+  sanitizeMetadataValue,
+  sanitizeAnalyticsTableValue,
+  isDefined,
+  hasOwnProperties,
+} from './utils';
 import { validateEventOptions, validateAnalyticsOptions } from './validation';
 import { sanitiseFetchDataOptions } from './sanitiseFetchDataOptions';
 
 const EVENT_DATE_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-const buildEventDataValues = resultsForEvent =>
-  resultsForEvent.reduce(
+const buildDataValuesFromAnswers = (answersForEvent: EventAnswer[]) =>
```

Comment:
Minor: similarly, this looks like a great rename for clarity. Love it, but would be easier to have this separate refactor be in a separate commit if possible.